src/compiler/diagnosticMessages.json

```diff
@@ -5237,6 +5233,22 @@
         "category": "Message",
         "code": 90038
     },
+    "Delete destructuring": {

```

Comment:
Is the change in language from "Remove *" to "Delete *" necessary?

Reply:
Basically, the logic related to action `Delete destructuring` is the same as before except one case related to functions parameters.

_Before_ 
```ts
export {};
const { x, y }/*Delete destructuring*/ = o;
```

_After_
```ts
export {};
```
---
If some of the object destruction is used 

_Before_
```ts
export {};
const { x, y/* Remove unused declaration for: 'y' */ } = o;
console.log(x);
```

_After_
```ts
export {};
const { x } = o;
console.log(x);
```

---
The changes related only if destruction is a parameter

```ts
function foo({ x, y }/* 1. Delete destructuring elements
                        2. Delete destructuring parameter */, z: number) { return z };
```

After applying the first action
```ts
function foo({}, z: number) { return z };
```

After applying the second action
```ts
function foo(z: number) { return z };
```
---
Also, the important part related to FixAll action, 

````ts
export {}
const x = 1;
const { x1, x2 } = {}
function a({ x, y }/* 1. Delete all unused declarations
                      2. Delete all destructuring parameters */, z: number) { return z };
function b({ x, y }, z: number) { return z };
a({  }, 1);
b({  }, 2);
````

After applying the first action
```ts
export {}
function a({  }, z: number) { return z };
function b({  }, z: number) { return z };
a({  }, 1);
b({  }, 2);
```

After applying the second action
```ts
export {}
const x = 1;
const { x1, x2 } = {}
function a(z: number) { return z };
function b(z: number) { return z };
a(1);
b(2);
```
---
> Also I would expect "Delete destructuring elements" to be "Remove destructuring elements: {0}"

`Delete destructuring elements` applies only to case when all destructured elements are unused, I'm not sure that we need to list all elements in QF action.

---
It seems this issue it really tricky and requires discussion. Removing destructuring parameters is really unsafe, however, do we really need a new action to remove only destructuring parameters? Maybe it would be better to define a clear flow for this QF, and then apply changes... I'll switch PR to Draft, or if you prefer I'll close it in favor of other PRs.

Reply:
Thanks for the clarification on the behavior. Having a code action to delete unused destructuring parameters does seem questionable when removing those parameters is the behavior we want to avoid in the first place. However, since you are supplanting removing destructuring _parameters_ with removing destructuring _elements_ when `Delete all unused declarations` is invoked (which is the fix we want), it does seem reasonable to separately offer the old behavior if it would be used.

@orta you offered insight on the original issue so I would appreciate your opinion on when we need to delete elements vs parameters. Should we ever offer to delete the parameters themselves, or only the unused elements?

Reply:
Cool, looks good to republish.